src/io/flutter/run/daemon/DeviceService.java

```diff
@@ -159,6 +163,31 @@ private void refreshDeviceDaemon() {
     deviceDaemon.refresh(this::chooseNextDaemon);
   }
 
+  private void daemonStopped(String details) {
+    if (project.isDisposed()) return;
+
+    final DeviceDaemon current = deviceDaemon.getNow();
+    if (current == null || current.isRunning()) {
+      // The active daemon didn't die, so it must be some older process.
+      // Just log it.
+      LOG.info("A Flutter device daemon stopped.\n" + details);
+      return;
+    }
+
+    // If we haven't tried restarting recently, try again.
+    final long now = System.currentTimeMillis();
+    final long millisSinceLastRestart = now - lastRestartTime.get();
+    if (millisSinceLastRestart > TimeUnit.MINUTES.toMillis(5)) {
+      LOG.info("A Flutter device daemon stopped. Automatically restarting it.\n" + details);
+      refreshDeviceDaemon();
+      lastRestartTime.set(now);
+      return;
+    }
+
+    // Report this so the user can send it to us, without restarting.
+    LOG.error("A Flutter device daemon stopped unexpectedly.\n" + details);
```

Comment:
I believe you could also do `LOG.error("A Flutter device daemon stopped unexpectedly", details);`, which would pass the 2nd arg through as a 'details' param.